### ISSUE TITLE
[Repo DX]: Add a GitHub action to check if files are prettified to ensure code cleanliness

### ISSUE THREAD

**Issue by Just-Moh-it** [remotion-dev/remotion#1858]
### Why? 🤔

Just saw a few pull requests with un-prettified code and messed up indentation. Another repo I worked on that had a prettier check, and it suddenly clicked in my mind.

We could have a GitHub action that could check if all files are prettified or not, and if they aren't, it'd fail the check and ask the user to commit after running a prettier script.

The prettify script would simply be `prettier -w .` and that'd prettify all code in the repo.

Let me know what you think! Feel free to share any wrong assumptions I'm making.

cc @JonnyBurger 🙃

**Comment by JonnyBurger** [comment]
A good idea.. Some nuances which can cause some complications:

A lot of PRs are open which will have to be rebased, so maybe it would make sense to run prettier over it for or after a big release, like the upcoming v4 release

I tried a GitHub action before that fixed all ESLint+Prettier issue and a bot would commit whenever you make a mistake

Problem is often you don't pull and a new commit gets added to your branch, and then a merge conflict emerges and I stopped doing it because it became annoying

**Comment by Just-Moh-it** [comment]
> A lot of PRs are open which will have to be rebased, so maybe it would make sense to run prettier over it for or after a big release, like the upcoming v4 release

@JonnyBurger yeah, would be a good idea to do a widespread prettify with v4

> Problem is often you don't pull and a new commit gets added to your branch, and then a merge conflict emerges and I stopped doing it because it became annoying

Totally agree. According to my knowledge, we ran into the same issue in our org. The solution was to not have a bot commit to prettify it, since that would result in conflicts. Rather, just have an action that checks prettier status and fails if it is not prettified. So the user has to manually run prettier and as a result, they are the only authors of the PR and no conflicts 🙃. This worked out really well for us, and so we kept it. Let me know if this approach works?
